modules/core/src/main/java/org/apache/ignite/internal/commandline/cache/CheckIndexInlineSizes.java

```diff
@@ -0,0 +1,176 @@
+            Collection<UUID> supportedServerNodeIds = F.transform(supportedServerNodes, GridClientNode::nodeId);
+
+            CheckIndexInlineSizesResult res = client.compute().projection(supportedServerNodes).execute(
+                CheckIndexInlineSizesTask.class.getName(),
+                new VisorTaskArgument<>(supportedServerNodeIds, false)
+            );
+
+            Set<UUID> unsupportedNodes = serverNodeIds.stream()
+                .filter(n -> !supportedServerNodeIds.contains(n))
+                .collect(toSet());
+
+            analizeResults(log, unsupportedNodes, res);
+        }
+
+        return null;
+    }
+
+    /** */
+    private void analizeResults(
+        Logger log,
+        Set<UUID> unsupportedNodes,
+        CheckIndexInlineSizesResult res
+    ) {
+        if (!F.isEmpty(unsupportedNodes))
+            log.info(String.format(NOT_ALL_NODES_SUPPORT_FEATURE_WARN_MSG_FMT, unsupportedNodes));
+
+        Map<String, Map<Integer, Set<UUID>>> indexToSizeNode = new HashMap<>();
+
+        for (Map.Entry<UUID, Map<String, Integer>> nodeRes : res.inlineSizes().entrySet()) {
+            for (Map.Entry<String, Integer> index : nodeRes.getValue().entrySet()) {
+                Map<Integer, Set<UUID>> sizeToNodes = indexToSizeNode.computeIfAbsent(index.getKey(), x -> new HashMap<>());
+
+                sizeToNodes.computeIfAbsent(index.getValue(), x -> new HashSet<>()).add(nodeRes.getKey());
+            }
+        }
+
+        log.info("Found " + indexToSizeNode.size() + " secondary indexes.");
+
+        Map<String, Map<Integer, Set<UUID>>> problems = indexToSizeNode.entrySet().stream()
+            .filter(e -> e.getValue().size() > 1)
+            .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+
+        if (F.isEmpty(problems))
+            log.info(INDEXES_INLINE_SIZE_ARE_SAME);
+        else
+            printProblemsAndShowRecommendations(problems, log);
+    }
+
+    /** */
+    private void printProblemsAndShowRecommendations(Map<String, Map<Integer, Set<UUID>>> problems, Logger log) {
+        log.info(problems.size() +
+            " indexes have different effective inline size on nodes. It can lead to performance degradation in SQL queries.");
+        log.info("Indexes:");
+
+        for (Map.Entry<String, Map<Integer, Set<UUID>>> entry : problems.entrySet()) {
+            SB sb = new SB();
+
+            sb.a("Full index name: ").a(entry.getKey());
+
+            for (Integer size : entry.getValue().keySet())
+                sb.a(" nodes: ").a(entry.getValue().get(size)).a(" inline size: ").a(size).a(",");
+
+            sb.setLength(sb.length() - 1);
+
+            log.info(INDENT + sb);
+        }
+
+        log.info("");
+
+        log.info("Recommendations:");
+        log.info(INDENT + "Check that value of property " + IgniteSystemProperties.IGNITE_MAX_INDEX_PAYLOAD_SIZE + " are the same on all nodes.");
```

Comment:
1. which low or high inline size need to be applied ? 
2. can u describe recreation procedure ? may be more user friendly here ?

Reply:
>which low or high inline size need to be applied ?

It depends on the user's choice. The property disabled by default, if the user set it you made some calculations and he knows right property value.

Reply:
> can u describe recreation procedure ? may be more user friendly here ?

Done